src/machine/machine-dbus.c

```diff
@@ -1002,16 +1000,11 @@ int bus_machine_method_bind_mount(sd_bus_message *message, void *userdata, sd_bu
                         goto child_fail;
                 }
 
-                if (make_file_or_directory) {
-                        if (S_ISDIR(st.st_mode))
-                                (void) mkdir_p(dest, 0755);
-                        else {
-                                (void) mkdir_parents(dest, 0755);
-                                (void) mknod(dest, S_IFREG|0600, 0);
-                        }
-                }
-
                 mount_inside = strjoina("/run/host/incoming/", basename(mount_outside));
+
+                if (make_file_or_directory)
+                        (void) make_mount_point_inode(mount_inside, dest, 0700);
```

Comment:
Is this correct? Previously, `st` was for `chased_src`. But now it looks `mount_inside`.

Reply:
That was a shortcut, mount_inside would be a mountpoint of the correct type at that moment, since immediately after it's the source of the mount syscall. But it's been reworked now.